homeassistant/helpers/entity_component.py

```diff
@@ -126,40 +180,56 @@ def add_entity(self, entity, platform=None):
                 object_id = '{} {}'.format(platform.entity_namespace,
                                            object_id)
 
-            entity.entity_id = generate_entity_id(
+            entity.entity_id = async_generate_entity_id(
                 self.entity_id_format, object_id,
                 self.entities.keys())
 
         self.entities[entity.entity_id] = entity
-        entity.update_ha_state()
+        self.hass.loop.create_task(entity.async_update_ha_state())
```

Comment:
This is breaking on the promise that when `async_add_entity` gets called (and thus also when `add_entities`) gets called that the state has not been written to the state machine yet (because `async_update_ha_state` will run after this task is done).

We should make sure to wait till `async_update_ha_state` is done. So:
 - Convert `async_add_entity` to a coroutine
 - `yield from entity.async_update_ha_state()`
 - Convert `async_add_entities` to a coroutine
 - `add_entities` should use `run_coroutine_threadsafe`